### ISSUE TITLE
memoizing inverse Vandermonde ignores possibly varying tind

### ISSUE THREAD

**Issue by gdmcbain** [kinnala/scikit-fem#68]
Investigating why I was getting seemingly different exceptions in #62 and couldn't work out the shape needed for `X` in `ElementH2.gbasis` there, I noticed a possibly dangerous memoization:

```python
V = None
```
and
```
       if self.V is None:
            # construct Vandermonde matrix and invert it
            self.V = np.linalg.inv(self._eval_dofs(mapping.mesh, tind=tind))
```
but what if `tind` (an optional keyword-argument of `gbasis`) changes?

**Comment by gdmcbain** [comment]
Indeed, running ex18 through `pdb`, I found that `ib.elem.V` is already defined before interpolation is attempted, with `ib.elem.V.shape == (551, 6, 6)`.  Here `mesh.t.shape[1] == 551` and `ib.Nbfun == 6`.

**Comment by kinnala** [comment]
Indeed, a quick workaround would be to memoize all elements and then slice the resulting V.

This is slightly non-optimal in FacetBasis, where only boundary elements are needed. So maybe we should add another optional flag, which disables the memoization of all elements, and enable the flag in the constructor of FacetBasis.

**Comment by kinnala** [comment]
An attempt:

[fix.txt](https://github.com/kinnala/scikit-fem/files/2456923/fix.txt)

**Comment by kinnala** [comment]
Ok, I'm feeling lucky. I merged this in 466049c73a1b7b0fd1f3b2f185465b6c7f6a4895 together with an implementation of `InteriorBasis.interpolator`. Let's hope it works.

**Comment by kinnala** [comment]
The proposed optimisation flag is not there, but I suppose the facet assembly on H2 elements is not a common bottleneck, so I'll close this.
